src/init/features/apphosting/index.ts

```diff
@@ -136,6 +157,69 @@ export async function createBackend(
     pollerName: `create-${projectId}-${location}-${backendId}`,
     operationResourceName: op.name,
   });
-
   return backend;
 }
+
+/**
+ * Only lowercase, digits, and hyphens; must begin with letter, and cannot end with hyphen
+ */
+function generateBuildId(n = 6) {
+  const letters = "abcdefghijklmnopqrstuvwxyz";
+  const allChars = "01234567890-abcdefghijklmnopqrstuvwxyz";
+  let id = letters[Math.floor(Math.random() * letters.length)];
+  for (let i = 1; i < n; i++) {
+    const idx = Math.floor(Math.random() * allChars.length);
+    id += allChars[idx];
+  }
+  return id;
```

Comment:
uuid?

Reply:
that could work except i don't think uuid's are guaranteed to start with non-number